app/controllers/models_controller.rb

```diff
@@ -6,6 +6,36 @@ class ModelsController < ApplicationController
 
   def index
     @models = resource_base_search_and_page
+    @hosts_counts = Model.hosts_counts
+    @table = @models.map do |m|
+      {
+        :id => m.id,
+        :name => m.name,
+        :vendor_class => m.vendor_class,
+        :hardware_model => m.hardware_model,
+        :hosts_count => @hosts_counts.fetch(m.id, 0),
+        :actions => m.actions_by_user(User.current)
+      }
+    end
+    sort_param = params.fetch(:order, "").split
+    by = (sort_param.length == 2) ? sort_param[0] : nil
+    order = (sort_param.length == 2) ? sort_param[1] : nil
+    @json_models = {
+      :results => @table,
+      :per_page => params.fetch(:per_page, "20"),
```

Comment:
the defaults should not be defined here, but rather reuse settings.

Reply:
I agree with you. I am going to look how to wrap the `@table` response in a way it's done in the `api/v2` so the defaults will be taken from the `api/v2`.